From b2df1ede9493c91d9b05c75e4770b732a26e5004 Mon Sep 17 00:00:00 2001
From: Pino Toscano <toscano.pino@tiscali.it>
Date: Sat, 28 Nov 2020 11:18:35 +0100
Subject: [PATCH 1/3] Define OSMINOR & OS_HURD on GNU/Hurd

Add a way to identify GNU/Hurd with b2, and also in the Python support.
---
 src/engine/jam.h     | 4 ++++
 src/tools/builtin.py | 6 +++---
 2 files changed, 7 insertions(+), 3 deletions(-)

Index: boost/tools/build/src/engine/jam.h
===================================================================
--- boost.orig/tools/build/src/engine/jam.h
+++ boost/tools/build/src/engine/jam.h
@@ -189,6 +189,10 @@
     #define OSMINOR "OS=DGUX"
     #define OS_DGUX
 #endif
+#ifdef __GNU__
+    #define OSMINOR "OS=HURD"
+    #define OS_HURD
+#endif
 #ifdef __hpux
     #define OSMINOR "OS=HPUX"
     #define OS_HPUX
Index: boost/tools/build/src/tools/builtin.py
===================================================================
--- boost.orig/tools/build/src/tools/builtin.py
+++ boost/tools/build/src/tools/builtin.py
@@ -82,9 +82,9 @@ def variant (name, parents_or_properties
     feature.compose ("<variant>" + name, explicit_properties.all())
 
 __os_names = """
-    amiga aix appletv bsd cygwin darwin dos emx freebsd hpux iphone linux netbsd
-    openbsd osf qnx qnxnto sgi solaris sun sunos svr4 sysv ultrix unix unixware
-    vms windows
+    amiga aix appletv bsd cygwin darwin dos emx freebsd hpux hurd iphone linux
+    netbsd openbsd osf qnx qnxnto sgi solaris sun sunos svr4 sysv ultrix unix
+    unixware vms windows
 """.split()
 
 # Translates from bjam current OS to the os tags used in host-os and target-os,
Index: boost/tools/build/src/engine/jam.cpp
===================================================================
--- boost.orig/tools/build/src/engine/jam.cpp
+++ boost/tools/build/src/engine/jam.cpp
@@ -747,7 +747,7 @@ char * executable_path( char const * arg
     sysctl( mib, 4, buf, &size, NULL, 0 );
     return ( !size || size == sizeof( buf ) ) ? NULL : strndup( buf, size );
 }
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__) || defined(__GNU__)
 # include <unistd.h>
 char * executable_path( char const * argv0 )
 {
Index: boost/tools/build/src/engine/sysinfo.cpp
===================================================================
--- boost.orig/tools/build/src/engine/sysinfo.cpp
+++ boost/tools/build/src/engine/sysinfo.cpp
@@ -18,7 +18,7 @@
 #include <unistd.h>
 #endif
 
-#if defined(OS_LINUX)
+#if defined(OS_LINUX) || defined(__GLIBC__)
 // Need to define this in case it's not as that's the only way to get the
 // sched_* APIs.
 #ifndef _GNU_SOURCE
